File: src/engine/commands/saycommand.cpp
#include "saycommand.h"


#define super Command

SayCommand::SayCommand(QObject *parent) :
    super(parent) {

    setDescription("Say something. Everyone in the same room can hear it.\n"
                   "\n"
                   "Example: say How are you?");
}

SayCommand::~SayCommand() {
}

void SayCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    if (!assertWordsLeft("Say what?")) {
        return;
    }

    QString message = takeRest();

    player->say(message);
}

File: data/commands/saycommand.js
function SayCommand() {
    Command.call(this);

    this.setDescription("Say something. Everyone close to you can hear it.\n" +
                        "\n" +
                        "Example: say How are you?");
}

SayCommand.prototype = new Command();
SayCommand.prototype.constructor = SayCommand;

SayCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    if (!this.assertWordsLeft("Say what?")) {
        return;
    }

    var message = this.takeRest();

    player.say(message);
};

CommandRegistry.registerCommand("say", new SayCommand());
